arrows/qt/widgets/MetadataView.cxx

```diff
@@ -30,135 +30,150 @@ namespace kwiver {
-      l->setToolTip({});
-      l->setEnabled(false);
+      l->setText( "(empty)" );
+      l->setToolTip( {} );
+      l->setEnabled( false );
     }
     else
     {
-      l->setText(valueText.trimmed(), qtSqueezedLabel::SetToolTip);
-      l->setEnabled(true);
+      l->setText( valueText.trimmed(), qtSqueezedLabel::SetToolTip );
+      l->setEnabled( true );
     }
   }
 }
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::clearItemValue(int id)
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::clearItemValue( int id )
 {
-  if (auto* const l = this->m_valueLabels.value(id, nullptr))
+  if ( auto* const l = this->m_valueLabels.value( id, nullptr ) )
   {
-    l->setText("(not available)");
-    l->setToolTip({});
-    l->setEnabled(false);
+    l->setText( "(not available)" );
+    l->setToolTip( {} );
+    l->setEnabled( false );
   }
 }
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::removeItem(int id)
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::removeItem( int id )
 {
-  delete this->m_keyLabels.take(id);
-  delete this->m_valueLabels.take(id);
+  delete this->m_keyLabels.take( id );
+  delete this->m_valueLabels.take( id );
 }
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::clear()
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::clear()
 {
-  qDeleteAll(this->m_keyLabels);
-  qDeleteAll(this->m_valueLabels);
+  qDeleteAll( this->m_keyLabels );
+  qDeleteAll( this->m_valueLabels );
   this->m_keyLabels.clear();
   this->m_valueLabels.clear();
 }
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::setKeyLabelColor(QLabel* label)
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::setKeyLabelColor( QLabel* label )
 {
   auto const& p = label->palette();
-  auto const& bg = p.color(QPalette::Window);
-  auto const& fg = p.color(QPalette::WindowText);
-  auto const& c = qtColorUtil::blend(bg, fg, 0.6);
-  label->setStyleSheet(QString{"color: %1;"}.arg(c.name()));
+  auto const& bg = p.color( QPalette::Window );
+  auto const& fg = p.color( QPalette::WindowText );
+  auto const& c = qtColorUtil::blend( bg, fg, 0.6 );
+  label->setStyleSheet( QString{ "color: %1;" }.arg( c.name() ) );
 }
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::updateLabelColors()
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::updateLabelColors()
 {
-  for (auto* const l : this->m_keyLabels)
+  for ( auto* const l : this->m_keyLabels )
```

Comment:
My C++ weakness is showing. Would `auto const* l` be equivalent here? If so, is that recommended given we use `auto const& l`?

Reply:
No. Always read types right to left:
- `auto* const` → immutable pointer to auto
- `auto const*` → pointer to immutable auto

These are very different. Consider:

| Operation | `auto* const` | `auto const*` |
| ---- | ---- | ---- |
| `p->nonConstMember();` | okay | error |
| `p = nullptr;` | error | okay |

Also note:
- `auto const* const` → immutable pointer to immutable auto
- `auto const&` → reference (which is implicitly immutable) to immutable auto

The general rule is prefer to give locals a terminal (right-most) `const` if they are not going to be mutated (which is almost always the case in range-based `for`), and prefer to also make `const` anything else you can make `const`.

Reply:
To add some context, note that `setKeyLabelColor` takes a `QLabel*`, not a `QLabel const*`, so `auto const*` would result in non-compiling code.

Reply:
Thank you, much appreciated.

Reply:
As an aside, this is related to why I have a rather strong preference for `T const` over `const T`. The former follows the usual rule that "`const` modifies whatever it immediately follows", while the latter is an exception to that rule. (The "main" reason is because if `T` is a type alias, you have to mentally rewrite as `T const` anyway or risk reading the type wrong... as in mistaking `const LPSTR` to be the same as `const char*`. I will refrain from naming the library that completely botched its C API due to that mistake.)